bundles/core/src/main/java/com/adobe/cq/wcm/core/components/internal/link/LinkImpl.java

```diff
@@ -49,16 +50,51 @@
     }};
 
     private final String url;
+    private final String mappedUrl;
     private final T reference;
-    private final LinkProcessor linkProcessor;
     private final Map<String, String> htmlAttributes;
-    private String processedUrl;
-    private boolean processed = false;
+    private final String externalizedUrl;
 
-    public LinkImpl(String url, T reference, LinkProcessor linkProcessor, Map<String, String> htmlAttributes) {
+    /**
+     * @param url Link URL
+     */
+    public LinkImpl(String url) {
```

Comment:
Is there a reason for introducing all these constructors? I don't see them being used (outside tests) and we could do without.